packages/@aws-cdk/aws-servicecatalog/lib/private/association-manager.ts

```diff
@@ -168,7 +195,30 @@ export class AssociationManager {
     return `- Portfolio: ${portfolio.node.path} | Product: ${product.node.path}`;
   }
 
+  private static formatTemplateRule(scope: cdk.Stack, rule: TemplateRule): string {
+    return JSON.stringify({
+      [rule.ruleName]: {
+        Assertions: this.formatAssertions(scope, rule.assertions),
+        RuleCondition: rule.condition ? scope.resolve(rule.condition) : undefined,
+      },
+    });
+  }
+
+  private static formatAssertions(scope: cdk.Stack, assertions : TemplateRuleAssertion[]): {[key: string]: string | undefined }[] {
+    const formattedAssertions: {[key: string]: string | undefined }[] = [];
+    assertions.reduce((formattedAssertion, assertion) => {
+      formattedAssertion.push( {
+        Assert: scope.resolve(assertion.assert),
+        AssertDescription: assertion.description,
+      });
+      return formattedAssertion;
+    }, formattedAssertions);
+
+    return formattedAssertions;
```

Comment:
No need for the `formattedAssertions` variable - `reduce` will return the value once it's done.